crates/bevy_transform/src/components/transform.rs

```diff
@@ -1,167 +1,132 @@
+
         Transform {
-            value: Mat4::identity(),
+            translation,
+            rotation,
+            scale,
         }
     }
 
     pub fn from_translation(translation: Vec3) -> Self {
-        Transform::new(Mat4::from_translation(translation))
+        Transform {
+            translation,
+            ..Default::default()
+        }
     }
 
     pub fn from_rotation(rotation: Quat) -> Self {
-        Transform::new(Mat4::from_quat(rotation))
+        Transform {
+            rotation,
+            ..Default::default()
+        }
     }
 
     pub fn from_scale(scale: f32) -> Self {
-        Transform::new(Mat4::from_scale(Vec3::splat(scale)))
+        Transform {
+            scale: Vec3::one() * scale,
+            ..Default::default()
+        }
     }
 
-    pub fn from_translation_rotation(translation: Vec3, rotation: Quat) -> Self {
-        Transform::new(Mat4::from_scale_rotation_translation(
-            Vec3::splat(1.0),
-            rotation,
-            translation,
-        ))
+    pub fn from_non_uniform_scale(scale: Vec3) -> Self {
+        Transform {
+            scale,
+            ..Default::default()
+        }
     }
 
-    pub fn from_translation_rotation_scale(translation: Vec3, rotation: Quat, scale: f32) -> Self {
-        Transform::new(Mat4::from_scale_rotation_translation(
-            Vec3::splat(scale),
-            rotation,
+    pub fn from_translation_rotation(translation: Vec3, rotation: Quat) -> Self {
+        Transform {
             translation,
-        ))
-    }
-
-    pub fn from_non_uniform_scale(scale: Vec3) -> Self {
-        Transform::new(Mat4::from_scale(scale))
+            rotation,
+            ..Default::default()
+        }
     }
 
     pub fn with_translation(mut self, translation: Vec3) -> Self {
-        self.set_translation(translation);
+        self.translation = translation;
         self
     }
 
     pub fn with_rotation(mut self, rotation: Quat) -> Self {
-        self.set_rotation(rotation);
+        self.rotation = rotation;
         self
     }
 
     pub fn with_scale(mut self, scale: f32) -> Self {
-        self.set_scale(scale);
+        self.scale = Vec3::one() * scale;
         self
     }
 
     pub fn with_non_uniform_scale(mut self, scale: Vec3) -> Self {
-        self.set_non_uniform_scale(scale);
-        self
-    }
-
-    pub fn with_translate(mut self, translation: Vec3) -> Self {
-        self.translate(translation);
-        self
-    }
-
-    pub fn with_rotate(mut self, rotation: Quat) -> Self {
-        self.rotate(rotation);
+        self.scale = scale;
         self
     }
 
-    pub fn with_apply_scale(mut self, scale: f32) -> Self {
-        self.apply_scale(scale);
-        self
+    /// Returns transform with the same translation and scale, but rotation so that transform.forward() points at the origin
+    pub fn looking_at_origin(self) -> Self {
+        self.looking_at(Vec3::zero(), Vec3::unit_y())
```

Comment:
As an aside it might pay for bevy to have some constants for `UP`, `FORWARD` and `SIDE` (or `RIGHT`) axes, might make it clearer what the coordinate system is.

Reply:
I've added support for creating constants in `glam` but I need to publish a new release. It's a bit tricky because underlying intrinsics aren't `const fn` so macros have to be used.

Reply:
Ooh thats cool / much appreciated :smile: